util/domainutil/domain_vars.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package domainutil
+
+import (
+	"strings"
+	"sync/atomic"
+
+	"github.com/pingcap/parser/model"
+)
+
+type repairInfo struct {
+	repairMode      atomic.Value
+	repairTableList atomic.Value
+	repairDBInfoMap atomic.Value
+}
+
+// RepairInfo indicates the repaired table info.
+var RepairInfo repairInfo
+
+// InRepairMode get whether TiDB is in repairMode.
+func (r *repairInfo) GetRepairMode() bool {
+	return r.repairMode.Load().(bool)
+}
+
+// InRepairMode set whether TiDB is in repairMode.
+func (r *repairInfo) SetRepairMode(mode bool) {
+	r.repairMode.Store(mode)
+}
+
+// InRepairMode tet the simple repaired table list.
+func (r *repairInfo) GetRepairTableList() []string {
+	return r.repairTableList.Load().([]string)
+}
+
+// InRepairMode set the simple repaired table list.
+func (r *repairInfo) SetRepairTableList(list []string) {
+	for i, one := range list {
+		list[i] = strings.ToLower(one)
+	}
+	r.repairTableList.Store(list)
+}
+
+// GetTablesInRepair return the map of repaired table in repair.
+func (r *repairInfo) GetTablesInRepair() map[int64]*model.DBInfo {
+	return r.repairDBInfoMap.Load().(map[int64]*model.DBInfo)
+}
+
+// GetRepairCleanFunc return a func for call back when repair action done.
+func (r *repairInfo) GetRepairCleanFunc() func(a, b string) {
+	return r.RemoveFromRepairList
+}
+
+// FetchRepairedTableList fetch the repaired table list from meta.
+func (r *repairInfo) FetchRepairedTableList(di *model.DBInfo, tbl *model.TableInfo) bool {
+	if r.repairMode.Load().(bool) {
+		isRepair := false
+		ls := r.repairTableList.Load().([]string)
+		for _, tn := range ls {
+			// Use dbName and tableName to specified a table.
+			if strings.ToLower(tn) == di.Name.L+"."+tbl.Name.L {
+				isRepair = true
+				break
+			}
+		}
+		if isRepair {
+			mp := r.repairDBInfoMap.Load().(map[int64]*model.DBInfo)
+			// Record the repaired table in Map.
+			if repairedDB, ok := mp[di.ID]; ok {
+				repairedDB.Tables = append(repairedDB.Tables, tbl)
+			} else {
+				// Shallow copy the DBInfo.
+				repairedDB := di.Copy()
+				// Clean the tables and set repaired table.
+				repairedDB.Tables = []*model.TableInfo{}
+				repairedDB.Tables = append(repairedDB.Tables, tbl)
```

Comment:
Why do we need to copy it? If it is needed. I think we can write it as follows:
```
repairedDB.Tables = []*model.TableInfo{tbl}
```

Reply:
Good catch!